cartoframes/dataset.py

```diff
@@ -243,6 +247,50 @@ def get_table_column_names(self, exclude=None):
 
         return columns
 
+    def compute_geom_type(self):
+        """Compute the geometry type from the data"""
+
+        if self.state == Dataset.STATE_REMOTE:
+            if self.query:
+                return self._get_remote_geom_type(self.query)
+            elif self.table_name and self.schema:
+                query = 'SELECT * FROM "{0}"."{1}"'.format(self.schema, self.table_name)
+                return self._get_remote_geom_type(query)
+
+        elif self.state == Dataset.STATE_LOCAL:
+            if self.gdf is not None:
+                return self._get_local_geom_type(self.gdf)
+
+    def _get_remote_geom_type(self, query):
+        """Fetch geom type of a remote table"""
+        if self.cc:
+            response = self.cc.sql_client.send('''
+                SELECT distinct ST_GeometryType(the_geom) AS geom_type
+                FROM ({}) q
+                LIMIT 5
+            '''.format(query))
+            if response and response.get('rows') and len(response.get('rows')) > 0:
+                st_geom_type = response.get('rows')[0].get('geom_type')
+                if st_geom_type:
+                    return self._map_geom_type(st_geom_type[3:])
+
+    def _get_local_geom_type(self, gdf):
+        """Compute geom type of the local dataframe"""
+        if len(gdf.geometry) > 0:
+            geom_type = gdf.geometry[0].type
+            if geom_type:
+                return self._map_geom_type(geom_type)
+
+    def _map_geom_type(self, geom_type):
+        return {
+            'Point': Dataset.GEOM_TYPE_POINT,
```

Comment:
Only a question: why don't we use the usual ones (without the _map_geom_type method)
- Point
- MultiPoint
- LineString
- MultiLineString
- Polygon
- MultiPolygon

Reply:
Because we need to group to point, line and polygon. Point and MultiPoint have the same style, LineString and MultiLineString have the same style, etc.

You can take a look at how default styles are defined: https://github.com/CartoDB/cartoframes/pull/675/files#diff-3233096e59fdf4159c753178da9e9c37R31.

Reply:
I can move this function to the Source class.

Reply:
Maybe we should use this dict closer to the visualization part, but I think is good enough right now.